client/src/legacy/TinyMCE_sslink-email.js

```diff
@@ -59,8 +59,9 @@ jQuery.entwine('ss', ($) => {
       const handleHide = () => this.close();
       const handleInsert = (...args) => this.handleInsert(...args);
       const attrs = this.getOriginalAttributes();
-      const selection = tinymce.activeEditor.selection;
-      const selectionContent = selection.getContent() || '';
+      const editor = this.getElement().getEditor();
+      const selection = editor.getInstance().selection;
+      const selectionContent = editor.getSelection();
```

Comment:
```suggestion
      let selectionContent = selection.getContent() || selection.getNode() || '';
      if (typeof selectionContent === 'object') {
        selectionContent = selectionContent.outerHTML || '';
      }
```
Try to get just the text, if there is just text.
If there's no text, try to get the node and grab its outer HTML.

Note that I haven't tested this with anything other than an image so far. Make sure to double check how this interacts with things like iframes (when explicitly allowed), embeds, etc - make sure we're still only allowing links to be added to things that should be allowed.

Reply:
This solution will not work in the opposite direction if the user has not selected anything, since getNode returns the parent element and, accordingly, true.